enhancements/SEP75-automatic-workload-migration.md

```diff
@@ -0,0 +1,643 @@
+- Verify operator memory usage remains stable
+- Verify Kubernetes API rate limits not exceeded
+
+## Example Configuration
+
+### Basic Automatic Migration (Default - No Limits)
+```yaml
+apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: v1.26.0
+  updateStrategy:
+    type: RevisionBased
+    workloadMigration:
+      strategy: Batched
+      # batched.maxVersion is unset - workloads will migrate to any version automatically
+```
+
+### Patch-only Updates (Production with Strict Change Control)
+```yaml
+apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: v1.26.0
+  updateStrategy:
+    type: RevisionBased
+    workloadMigration:
+      strategy: Batched
+      batched:
+        maxVersion: "1.26.999"  # Only migrate within 1.26.x, stop at 1.27
+        batchSize: 3  # Conservative batching for production
+        delayBetweenBatches: 2m
+        readinessTimeout: 10m
+```
+
+### Controlled Minor Version Upgrades (Staged Rollout)
+```yaml
+apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: v1.26.0
+  updateStrategy:
+    type: RevisionBased
+    workloadMigration:
+      strategy: Batched
+      batched:
+        maxVersion: "1.26.0"  # Auto-migrate up to 1.26.x, then requires manual update
+        batchSize: 5
+        delayBetweenBatches: 60s
+        readinessTimeout: 10m
+```
+
+## Security Considerations
+
+There's a risk that we break security features configured by the user because they don't work properly in the new version, e.g. deprecated features or custom changes made using `EnvoyFilter`. We can't really mitigate this, so workload migration will always be disabled by default and should be used with caution.
+
+## Future Enhancements
+
+### Per-Workload Annotations
+
+While the initial implementation provides cluster-wide and namespace-level migration configuration through the `Istio` resource, some users may need to override migration behavior on a per-workload basis. This could be achieved through annotations on Deployment resources.
+
+**Potential annotations:**
+- `sailoperator.io/readiness-timeout: 10s`: Override the readiness timeout for specific workloads
+- `sailoperator.io/skip-migration: true`: Exclude specific workloads from automatic migration
+- `sailoperator.io/migration-batch: 3`: Assign workload to a specific migration priority (integer, lower values migrate first)
+
+### Failure Policy
+
+The initial implementation will continue processing all batches even if individual workloads fail to migrate, reporting all failures at the end. Some users may want different behavior when migrations fail. We could implement a `failurePolicy` field that allows users to specify what the operator should do in case a workload does not become ready.
+
+Potential `failurePolicy` values:
+* `ContinueOnError`: Continue migrating all workloads, report failures at end
+* `FailFast`: Stop immediately on first failure
```

Comment:
A configurable % of workloads that are allowed to fail before stopping the migration might be nice as it would give finer-grained control over success/failure conditions (i.e. "continue migration as long as no more than 5% of workloads failed upgrade").